packages/react-dom/src/shared/DOMProperty.js

```diff
@@ -139,56 +139,69 @@ export const ROOT_ATTRIBUTE_NAME = 'data-reactroot';
  */
 export const properties = {};
 
-/**
- * Checks whether a property name is a writeable attribute.
- * @method
- */
-export function shouldSetAttribute(name, value) {
+export function shouldSkipAttribute(name) {
   if (isReservedProp(name)) {
-    return false;
+    return true;
   }
   if (
     name.length > 2 &&
     (name[0] === 'o' || name[0] === 'O') &&
     (name[1] === 'n' || name[1] === 'N')
   ) {
-    return false;
-  }
-  if (value === null) {
     return true;
   }
+  return false;
+}
+
+export function isBadlyTypedAttributeValue(name, value) {
+  if (isReservedProp(name)) {
+    return false;
+  }
   switch (typeof value) {
-    case 'boolean':
-      return shouldAttributeAcceptBooleanValue(name);
-    case 'undefined':
-    case 'number':
-    case 'string':
-    case 'object':
+    case 'function':
+    case 'symbol':
       return true;
+    case 'boolean':
+      break;
     default:
-      // function, symbol
       return false;
   }
-}
-
-export function getPropertyInfo(name) {
-  return properties.hasOwnProperty(name) ? properties[name] : null;
-}
-
-export function shouldAttributeAcceptBooleanValue(name) {
-  if (isReservedProp(name)) {
-    return true;
-  }
   let propertyInfo = getPropertyInfo(name);
   if (propertyInfo) {
-    return (
+    return !(
       propertyInfo.hasBooleanValue ||
       propertyInfo.hasStringBooleanValue ||
       propertyInfo.hasOverloadedBooleanValue
     );
   }
   const prefix = name.toLowerCase().slice(0, 5);
-  return prefix === 'data-' || prefix === 'aria-';
+  return prefix !== 'data-' && prefix !== 'aria-';
+}
+
+export function shouldTreatAttributeValueAsNull(name, value) {
+  if (value === null || typeof value === 'undefined') {
```

Comment:
why do we need to do `typeof value === 'undefind'`? why not just `value === undefined`?

Reply:
Consistency with all other checks. (I don't care either way but IMO we should change them all together if we want to.)